ingestion/src/metadata/data_insight/processor/cost_analysis_report_data_processor.py

```diff
@@ -0,0 +1,234 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Processor class used to compute refined report data
+"""
+
+from __future__ import annotations
+
+import traceback
+from collections import defaultdict
+from typing import Iterable
+
+from metadata.data_insight.processor.data_processor import DataProcessor
+from metadata.generated.schema.analytics.reportData import ReportData, ReportDataType
+from metadata.generated.schema.analytics.reportDataType.aggregatedCostAnalysisReportData import (
+    AggregatedCostAnalysisReportData,
+)
+from metadata.generated.schema.analytics.reportDataType.rawCostAnalysisReportData import (
+    RawCostAnalysisReportData,
+)
+from metadata.generated.schema.entity.data import table
+from metadata.utils.logger import data_insight_logger
+from metadata.utils.time_utils import get_end_of_day_timestamp_mill
+
+logger = data_insight_logger()
+
+
+class RawCostAnalysisReportDataProcessor(DataProcessor):
+    """Processor class used as a bridge to refine the data"""
+
+    _data_processor_type = "RawCostAnalysisReportData"
+
+    def fetch_data(self) -> Iterable[table.Table]:
+        try:
+            yield from self.metadata.list_all_entities(
+                table.Table, limit=1000, fields=["*"]
```

Comment:
do we really need `*` in the fields? Maybe we can just ask for the minimum necessary ones to move less data, since we might already be pulling a bunch of tables